indico/web/client/js/custom_elements/ind_date_picker.js

```diff
@@ -387,175 +577,279 @@ customElements.define(
+              month: 'long',
+              year: 'numeric',
+            });
+          }
 
-function getFirstDayOfWeek(weekInfo, date) {
-  const offset = weekInfo.weekDays.indexOf(date.getDay() || 7);
-  const firstDayOfWeek = new Date(date);
-  firstDayOfWeek.setDate(firstDayOfWeek.getDate() - offset);
-  return firstDayOfWeek;
-}
+          const selectedRange = new DateRange(indDateGrid.rangeStart, indDateGrid.rangeEnd);
+
+          calendarButtons.forEach((calendarButton, i) => {
+            const date = new Date(firstDayOfCalendar);
+            date.setDate(date.getDate() + i);
+            const belongsToCurrentMonth = date.getMonth() === calendarMonth;
+            const isRenderable = !indDateGrid.hideDatesFromOtherMonths || belongsToCurrentMonth;
+
+            calendarButton.dataset.currentMonth = belongsToCurrentMonth;
+            calendarButton.disabled = !isRenderable;
+
+            if (!isRenderable) {
+              calendarButton.textContent = '';
+              calendarButton.removeAttribute('aria-label');
+              calendarButton.removeAttribute('aria-selected');
+              calendarButton.removeAttribute('data-week');
+              calendarButton.removeAttribute('data-range-start');
+              calendarButton.removeAttribute('data-range-end');
+              calendarButton.value = '';
+              calendarButton.tabIndex = -1;
+              calendarButton.disabled = true;
+              return;
+            }
+
+            // Update button attributes
+            calendarButton.textContent = date.getDate();
+            calendarButton.setAttribute(
+              'aria-label',
+              date.toLocaleDateString(indDateGrid.locale, {
+                year: 'numeric',
+                month: 'long',
+                day: 'numeric',
+              })
+            );
+            calendarButton.toggleAttribute(
+              'data-weekend',
+              weekInfo.weekend.includes(date.getDay() + 1)
+            );
+            calendarButton.tabIndex = -1;
+            calendarButton.value = toDateString(date);
+
+            if (this.allowableSelectionRange.includes(date)) {
+              calendarButton.removeAttribute('aria-disabled');
+            } else {
+              calendarButton.setAttribute('aria-disabled', true);
+            }
+
+            if (selectedRange.includes(date)) {
+              calendarButton.setAttribute('aria-selected', true);
+            } else {
+              calendarButton.removeAttribute('aria-selected');
+            }
+            calendarButton.toggleAttribute('data-range-start', selectedRange.startsWith(date));
+            calendarButton.toggleAttribute('data-range-end', selectedRange.endsWith(date));
+          });
+
+          const focusableButton = listbox.querySelector(
+            '[role=option]:is([aria-selected=true],[data-current-month=true]:not([aria-disabled]))'
+          );
+          focusableButton.tabIndex = 0;
```

Comment:
```
Uncaught TypeError: focusableButton is null
    debounceTimer ind_date_picker.js:820
    setTimeout handler*updateGridDates ind_date_picker.js:749
    attributeChangedCallback ind_date_picker.js:829
```

This happens e.g. when opening the date range picker in the accommodation field or when editing a room booking. It also happens when changing months/years.